python/pyarrow/array.pxi

```diff
@@ -3570,11 +3570,14 @@ cdef class FixedShapeTensorArray(ExtensionArray):
 
     def to_numpy_ndarray(self):
         """
-        Convert fixed shape tensor extension array to a numpy.ndarray with zero copy.
-        First dimension of ndarray will be the length of the fixed shape tensor array
+        Convert fixed shape tensor extension array to a multi-dimensional numpy.ndarray.
+
+        The resulting ndarray will have (ndim + 1) dimensions.
+        The size of the first dimension will be the length of the fixed shape tensor array
         and the rest of the dimensions will match the permuted shape of the fixed
         shape tensor.
 
+        The conversion is zero-copy.
```

Comment:
Small nit: this is only if the conversion to numpy is zero-copy (i.e. primitive numeric data without nulls)

Reply:
Good point, added to `VariableShapeTensor` PR https://github.com/apache/arrow/pull/38008/commits/8ca3bf7b36daf11c63a6dd665a9547997dfe3687